frontend/src/metabase/search/components/SearchSidebar/SearchSidebar.tsx

```diff
@@ -1,59 +1,104 @@
 import _ from "underscore";
+import { useCallback, useMemo } from "react";
 import type {
   FilterTypeKeys,
+  SearchFilterComponent,
   SearchFilterPropTypes,
   URLSearchFilterQueryParams,
-  SearchSidebarFilterComponent,
 } from "metabase/search/types";
 import { Stack } from "metabase/ui";
 import { SearchFilterKeys } from "metabase/search/constants";
-import { SidebarFilter } from "metabase/search/components/SidebarFilter/SidebarFilter";
+import { DropdownSidebarFilter } from "metabase/search/components/SearchSidebar/DropdownSidebarFilter/DropdownSidebarFilter";
 import { TypeFilter } from "metabase/search/components/filters/TypeFilter/TypeFilter";
+import { PLUGIN_CONTENT_VERIFICATION } from "metabase/plugins";
+import { ToggleSidebarFilter } from "metabase/search/components/SearchSidebar/ToggleSidebarFilter/ToggleSidebarFilter";
 import { CreatedByFilter } from "metabase/search/components/filters/CreatedByFilter/CreatedByFilter";
 
 type SearchSidebarProps = {
   value: URLSearchFilterQueryParams;
   onChange: (value: URLSearchFilterQueryParams) => void;
 };
 
-export const filterMap: Record<FilterTypeKeys, SearchSidebarFilterComponent> = {
-  [SearchFilterKeys.Type]: TypeFilter,
-  [SearchFilterKeys.CreatedBy]: CreatedByFilter,
-};
-
 export const SearchSidebar = ({ value, onChange }: SearchSidebarProps) => {
+  const filterMap: Record<FilterTypeKeys, SearchFilterComponent | null> =
```

Comment:
Could/should we move this map outside of the component so we don't need to use `useMemo` since it doesn't rely on any props?